packages/circuit-ui/components/Popover/Popover.spec.tsx

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright 2020, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Bin, CirclePlus, PenStroke, ThumbUp, Zap } from '@sumup/icons';
+import { fireEvent } from '@testing-library/dom';
+import React, { useRef, useState } from 'react';
+
+import {
+  create,
+  renderToHtml,
+  axe,
+  RenderFn,
+  render,
+  act,
+  userEvent,
+} from '../../util/test-utils';
+import Button from '../Button';
+
+import { PopoverItem, PopoverItemProps, Popover } from './Popover';
+
+const placements = ['auto', 'top', 'bottom', 'left', 'right'];
+
+describe('PopoverItem', () => {
+  function renderPopoverItem<T>(
+    renderFn: RenderFn<T>,
+    props: PopoverItemProps,
+  ) {
+    return renderFn(<PopoverItem {...props} />);
+  }
+
+  const baseProps = { children: 'PopoverItem' };
+
+  describe('styles', () => {
+    it('should render as Link when an href (and onClick) is passed', () => {
+      const props = {
+        ...baseProps,
+        href: 'https://sumup.com',
+        onClick: jest.fn(),
+        icon: Zap,
+      };
+      const actual = renderPopoverItem(create, props);
+      expect(actual).toMatchSnapshot();
+    });
+
+    it('should render as a `button` when an onClick is passed', () => {
+      const props = { ...baseProps, onClick: jest.fn(), icon: Zap };
+      const actual = renderPopoverItem(create, props);
+      expect(actual).toMatchSnapshot();
+    });
+  });
+
+  describe('accessibility', () => {
+    it('should meet accessibility guidelines', async () => {
+      const props = {
+        ...baseProps,
+        href: 'https://sumup.com',
+        onClick: jest.fn(),
+        icon: Zap,
+      };
+      const wrapper = renderPopoverItem(renderToHtml, props);
+      const actual = await axe(wrapper);
+      expect(actual).toHaveNoViolations();
+    });
+  });
+});
+
+describe('Popover', () => {
+  const Default = (props) => {
+    const referenceElement = useRef<HTMLButtonElement & HTMLAnchorElement>(
+      null,
+    );
+
+    return (
+      <>
+        <button ref={referenceElement}>Button</button>
+        <Popover referenceElement={referenceElement} {...props} />
+      </>
+    );
+  };
+
+  const Interactive = () => {
```

Comment:
@connor-baer thoughts on this? Since the component needs a ref, this is how we were thinking of testing it, although it's a bit verbose to write a component in here. It's also similar to what was done for the ImageInput

Reply:
It would be cool to somehow reuse these "test" components between the specs and the Storybook, because they're very similar

Reply:
I don't think the `Interactive` component is necessary. It adds overhead and you might end up testing the state logic, not the component interface. 

An example: To test if the component responds to outside clicks, I'd simply check if the `onClose` prop was called.

You could reuse the stories by simply importing them, although I'm not sure Jest is able to handle `.mdx` files. That could be configured though.

Reply:
Makes sense–I like the idea of clicking the button and asserting things like `getByTest("Popover content").toBeVisible()`, but it's true that this is basically component/PopperJS internals. So you would only keep the snapshots and a few logic tests for triggering the `onClose()` for example?

Reply:
Yes.